src/main.c

```diff
@@ -58,17 +59,15 @@ txn_approve()
   PRINTF("Signing message: %.*h\n", msg_len, msgpack_buf);
 
   cx_ecfp_private_key_t privateKey;
-  algorand_private_key(&privateKey);
+  algorand_key_derive(0, &privateKey);
```

Comment:
*Extension of my very 1st comment above*
Private key derivation is done on accountId 0. Does that mean this account is special to the app? In that case, does that mean accountId 0 cannot be passed as APDU payload (I didn't see a check for that in the APDU dispatcher code)? Anyway, a comment here would be welcome to explain shortly why accountId 0.

Reply:
@hcleonis yes you are right: that should be `current_txn.accountId` here.

Reply:
:indeed: tests should catch this, if that's not the case we need to add a test case.

Reply:
Fixed.